data-runtime/src/main/java/io/micronaut/data/runtime/query/internal/DefaultStoredQuery.java

```diff
@@ -115,7 +116,13 @@ public DefaultStoredQuery(
             Optional<String> rawCountQueryString = method.stringValue(Query.class, DataMethod.META_MEMBER_RAW_COUNT_QUERY);
             this.rawQuery = rawCountQueryString.isPresent();
             this.query = rawCountQueryString.orElse(query);
```

Comment:
This is the root cause, in `countBy` queries these expandable parts are empty and this is why binding didn't work.
That fixes it, not sure if there is some better fix but nothing broke so should be good:)